fossunited/foss_hackathon/doctype/foss_hackathon_localhost/foss_hackathon_localhost.py

```diff
@@ -35,55 +35,70 @@ class FOSSHackathonLocalHost(Document):
     pass
 
     def before_insert(self):
-        self.assign_localhost_organizer_role()
+        members = [member.profile for member in self.organizers]
+        self.assign_localhost_organizer_role(members)
 
-    def on_update(self):
-        self.check_if_member_removed()
+    def on_trash(self):
+        members = [member.profile for member in self.organizers]
+        self.remove_organizer_role(members)
 
     def before_save(self):
-        if self.has_value_changed("organizers") and (
-            len(self.organizers) > len(self.get_doc_before_save().organizers)
-        ):
-            self.assign_localhost_organizer_role()
+        self.handle_roles()
 
-    def assign_localhost_organizer_role(self):
-        for member in self.organizers:
+    def handle_roles(self):
+        if self.is_new():
+            return
+
+        organizers = self.get("organizers")
+        organizers_old = self.get_doc_before_save().get("organizers")
+
+        # get members that are in curr member but not in old members
+        new_members = [
+            member.profile
+            for member in organizers
+            if member.profile not in [m.profile for m in organizers_old]
+        ]
+        removed_members = [
+            member.profile
+            for member in organizers_old
+            if member.profile not in [m.profile for m in organizers]
+        ]
+
+        self.assign_localhost_organizer_role(new_members)
+        self.remove_organizer_role(removed_members)
+
+    def assign_localhost_organizer_role(self, members: list):
+        for member in members:
             user = frappe.get_doc(
                 "User",
-                frappe.db.get_value(USER_PROFILE, member.profile, "user"),
+                frappe.db.get_value(USER_PROFILE, member, "user"),
             )
             user.add_roles("Localhost Organizer")
```

Comment:
this is unrelated to this PR but we should add a constant for this - this is a string that has special meaning so we should make it easily discoverable like the IDs. if there are other roles, we could create a `fossunited.roles` module and add them there. we already define the `Localhost Organizer` string in the `remove_organizer_role` so this feels like a necessary change

nitpicky comment - `LocalHost` in the doctype vs `Localhost` in the role. I wish we had been thorough when we wrote this code originally.

Reply:
ignore - i've added a separate issue for this - https://github.com/fossunited/fossunited/issues/744 - to address all of the role constants